### DISCUSSION TITLE
make logs not publicly accessible

### DISCUSSION THREAD

**Discussion by sander1095** [jellyfin/jellyfin#9862] [General]
I looked into the logs of my server and found out you can access those without being logged in to the server. This is a huge security issue for me.

Why are they publicly accessible? I want to have to be logged in to view them..

**Comment by crobibero** [comment]
How are you able to access these logs without being logged in?

**Reply by sander1095** [reply]
I have a very default installation, i never remember having changed any auth settings, esecially not about log access.

I use the web client and copy one of these log links from the logs dashboard.

![Screenshot_20230606_220124_Firefox.jpg](https://github.com/jellyfin/jellyfin/assets/7312681/7c84329b-2e36-4b3b-ac8a-dc3edf55055d)

I open it in a private window (where i am not logged in) and then I see my logs.

**Reply by nielsvanvelzen** [reply]
Did you remove the API key from the URL? (The thing that tells the server who you are)

**Reply by Shadowghost** [reply]
I just tried it. The copied link includes the API key. Removing the latter denies access - access control works as intended.

**Reply by sander1095** [reply]
You're completely right. I didn't check the URL for a query parameter including the API key.
I do find that behaviour a bit odd. Things like this should be done in the `Authorization` header or using cookie auth. 

It being part of the query parameter has security implications:

- Caching
- API key is now saved in browser history
- it's non-standard

**Reply by Shadowghost** [reply]
We do need to use API keys as query parameter for some endpoints due to DLNA but most is just inherited code from Emby/MediaBrowser. We are working on switching the to something more "standard" for endpoints not used by DLNA.
